packages/venia-ui/lib/drivers/adapter.js

```diff
@@ -81,6 +81,25 @@ const VeniaAdapter = props => {
         })
     );
 
+    const [initialized, setInitialized] = useState(false);
+
+    useEffect(() => {
+        async function initialize() {
+            // On load, restore the persisted data to the apollo cache and then
+            // allow rendering. You can do other async blocking stuff here.
+            await persistor.restore();
+            setInitialized(true);
+        }
+        if (!initialized && persistor) {
+            initialize();
+        }
+    }, [initialized, persistor]);
+
+    if (!initialized) {
+        // TODO: Replace with app skeleton. See PWA-547.
+        return null;
+    }
```

Comment:
I broke this in #2322. We have to wait persistence otherwise we end up not restoring the cache from local storage which is counter to why we even include persistence layer.